lib/teleterm/clusters/cluster_access_requests.go

```diff
@@ -96,36 +78,25 @@ func (c *Cluster) CreateAccessRequest(ctx context.Context, req *api.CreateAccess
 		})
 	}
 
-	err = addMetadataToRetryableError(ctx, func() error {
-		proxyClient, err = c.clusterClient.ConnectToProxy(ctx)
-		if err != nil {
-			return trace.Wrap(err)
-		}
-		defer proxyClient.Close()
-
-		authClient, err = proxyClient.ConnectToCluster(ctx, c.clusterClient.SiteName)
-		if err != nil {
-			return trace.Wrap(err)
-		}
-		defer authClient.Close()
-
-		if len(req.Roles) > 0 {
-			request, err = services.NewAccessRequest(c.status.Username, req.Roles...)
-		} else {
-			request, err = services.NewAccessRequestWithResources(c.status.Username, nil, resourceIDs)
-		}
-		if err != nil {
-			return trace.Wrap(err)
-		}
+	// Role-based and Resource-based AccessRequests are mutually exclusive.
+	if len(req.Roles) > 0 {
+		request, err = services.NewAccessRequest(c.status.Username, req.Roles...)
+	} else {
+		request, err = services.NewAccessRequestWithResources(c.status.Username, nil, resourceIDs)
+	}
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 
-		request.SetRequestReason(req.Reason)
-		request.SetSuggestedReviewers(req.SuggestedReviewers)
+	request.SetRequestReason(req.Reason)
+	request.SetSuggestedReviewers(req.SuggestedReviewers)
 
-		if err := authClient.CreateAccessRequest(ctx, request); err != nil {
+	err = addMetadataToRetryableError(ctx, func() error {
+		if err := c.clusterClient.CreateAccessRequest(ctx, request); err != nil {
 			return trace.Wrap(err)
 		}
 
-		return nil
+		return err
```

Comment:
This can be replaced with `return nil` I think? We already handle the `err != nil` case above.

Reply:
You're right, but I might as well just change to 
```go
return c.clusterClient.CreateAccessRequest(ctx, request)
```
I think it's a bit easier to read. Thoughts?

Reply:
Yeah, that's fine.